### PR TITLE
Changes to fix non-reproducibility in MAMMicrophysics::run_impl.

### PR DESCRIPTION
**Pull request by overfelt**

### CODE REVIEW

File: src/mam4xx/mo_sethet.hpp [hunk 1]
```diff
@@ -295,21 +295,22 @@ void sethet_detail(
...
-    if (mm2 > 0) {
-      Kokkos::parallel_for(Kokkos::TeamVectorRange(team, local_pver),
-                           [&](int kk) { het_rates(kk, mm2) = MISSING; });
-    }
-  }
+  Kokkos::parallel_for(Kokkos::ThreadVectorRange(team, local_pver),
+                       [&](int kk) {
+                         for (int mm = 0; mm < gas_wetdep_cnt; mm++) {
+                           const int mm2 = wetdep_map[mm];
+                           if (mm2 > 0)
```
Comment (jaelynlitz) [inline src/mam4xx/mo_sethet.hpp:310]: @mjs271 and I discovered yesterday that this needs to be `>=` to account for fortran -> C++ indexing

Comment (overfelt) [reply src/mam4xx/mo_sethet.hpp]: Done.....

**Comment by jaelynlitz** [comment]
I see that `calc_precip_rescale` is failing in our Deception pipelines. I went to investigate by hand and I do not see the failure with the updated modules that are not in CI yet.

**Review by mjs271 — APPROVED** [review]
Looks good to me!

**Pull request merged**